### ISSUE TITLE
Strict bounds for many of the geometry parameters in the `HotRegion` class use `_pi = math.pi`.

### ISSUE THREAD

**Issue by DevarshiChoudhury** [xpsi-group/xpsi#332]
https://github.com/xpsi-group/xpsi/blob/b705e1772f2cd0d641551c49bb5d8f37ea02d083/xpsi/HotRegion.py#L290C57-L290C57

Instead of `math.pi` we usually preferably use `math.pi - 0.001` for the colatitude to avoid spots being centred on the poles thus leading to no signal modulation, and for angular radii it may be preferable to use `math.pi/2.0 - 0.001` since a spot that covers more than half the star is better modelled as elsewhere with a hole. Normally we set these manually. Could it perhaps be better to have them be the default bounds?

The only restriction then arises is if we want to assign spots these values specifically for testing purposes, but I feel these are edge cases. I think it's better to change it for general use and have the users/devs modify it themselves if they really need to.

If we decide to change this, then for colatitudes and `super` radii, the lower limits can then also be changed from 0.0 to 0.001.

**Comment by thjsal** [comment]
The choice of `0.001` seems a bit arbitrary to me though. Why not allow the user to set it to `0.00000001`? It is also not clear to me why the limit could not be exactly `0.0` (or `math.pi/2.0`), since during sampling it is very unlikely that you would get a sample precisely from the prior boundary, right?

Or if we are sure that this 0.001 is really the limit after which numerical problems arise, maybe we could indeed set the hard bounds to that.

**Comment by DevarshiChoudhury** [comment]
I agree that sampling `0.0` and `pi/2.0` exactly is highly unlikely, and indeed as long as the code doesn't break, it's fine to leave these values as is. I don't see why there'll be numerical problems if go down to `<0.001` if having a vallue of `0.0` works

**Comment by thjsal** [comment]
Should we then close this issue as "not planned"?
